Add to.dtype and neg ops

billmguo opened the pull request:
Summary: add neg ops and annonation of to.dtype

Differential Revision: D68815927

billmguo commented:
@pytorchbot label "topic: not user facing"

billmguo commented:
@shewu-quic  would u mind review and approve I will remove to.dypte ops since I modify model it can eliminate the to.dtype

shewu-quic commented on backends/qualcomm/tests/test_qnn_delegate.py:
```diff
@@ -682,6 +692,11 @@ def test_qnn_backend_view(self):
         sample_input = (torch.randn([1, 8, 512]), torch.randn([1, 2, 8, 256]))
         self.lower_module_and_test_output(module, sample_input)
 
+    def test_qnn_backend_argmin(self):
```
It seems to be duplicated

shewu-quic commented on backends/qualcomm/tests/utils.py:
```diff
@@ -125,12 +125,15 @@ class TestQNN(unittest.TestCase):
     image_dataset: str = ""
     pretrained_weight: str = ""
     enable_profile: bool = False
+    compile_only: bool = True
```
It seems to be duplicated

shewu-quic commented on backends/qualcomm/tests/utils.py:
```diff
@@ -125,12 +125,15 @@ class TestQNN(unittest.TestCase):
-    enable_x86_64: bool = False
+    enable_x86_64: bool = True
+    compile_only: bool = False
+    dump_intermediate_outputs: bool = False
```
May I know why do you need to add this variable?

billmguo replied:
this is chen's test code. I re-base on top her pr to run the unit test in x86 emulator

shewu-quic commented on backends/qualcomm/tests/utils.py:
```diff
@@ -197,6 +200,7 @@ def verify_output(  # noqa: C901
             )
 
             output_dir = f"{tmp_dir}/outputs"
+            os.makedirs(output_dir, exist_ok=True) 
```
make_output_dir(output_dir)

shewu-quic reviewed (commented):
@winskuo-quic, could you help to review this PR about CI

shewu-quic reviewed (approved):
LGTM. Thanks.

shewu-quic reviewed (approved):
LGTM. Thanks!

billmguo committed:
Add to.dtype and neg ops (#8041)

Summary:

https://github.com/pytorch/executorch/pull/8041
add neg ops and annonation of to.dtype

Reviewed By: Andriyluck

Differential Revision: D68815927
